src/usr/lib/rsetup/tui/overlay/overlay.sh

```diff
@@ -137,19 +137,30 @@ __overlay_manage() {
 
     disable_overlays
 
-    local item
+    local items ret
     for i in "${RSETUP_CHECKLIST_STATE_NEW[@]}"
     do
-        item="$(checklist_getitem "$i")"
-        mv "$U_BOOT_FDT_OVERLAYS_DIR/$item.disabled" "$U_BOOT_FDT_OVERLAYS_DIR/$item"
+        items+=("$(checklist_getitem "$i")")
     done
 
-    if u-boot-update >/dev/null
+    if [[ -z "${items[*]}" ]]
     then
-        msgbox "Selected overlays will be enabled at next boot."
-    else
-        msgbox "Unable to update the boot config."
+        msgbox "No overlay selected."
+        return
     fi
+
+    enable_overlays "${items[@]}"; ret="$?"
+    case "$ret" in
+        0)
+            msgbox "Selected overlays will be enabled at next boot."
+            ;;
+        "$ERROR_ILLEGAL_PARAMETERS")
+            msgbox "Have illegal parameters."
```

Comment:
```suggestion
            msgbox "The selection contains non-existing overlays. Did you delete them?"
```
错误消息要能够让人理解，尤其是这种你知道为什么报错的情况下